cli/flox/src/commands/generations/history.rs

```diff
@@ -40,13 +53,19 @@ impl History {
         let env: GenerationsEnvironment = env.try_into()?;
         let metadata = env.generations_metadata()?;
 
-        let output = DisplayHistory(metadata.history());
+        let output = match self.output_mode {
+            OutputMode::Pretty => DisplayHistory(metadata.history()).to_string(),
+            OutputMode::Json => {
+                serde_json::to_string_pretty(metadata.history()).expect("derived from valid json")
```

Comment:
nonblocking: I think having kind in the --json output feels a bit like leaking an implementation detail, but also probably won't hurt 🤷  We could filter it out though